Validation/HGCalValidation/test/HGCalWaferHitCheck.cc

```diff
@@ -0,0 +1,257 @@
+    } else {
+      HGCSiliconDetId detId = HGCSiliconDetId(id);
+      waferU = detId.waferU();
+      waferV = detId.waferV();
+      type = detId.type();
+      layer = detId.layer();
+      zside = detId.zside();
+    }
+    int index = HGCalWaferIndex::waferIndex(layer, waferU, waferV, false);
+    int typef = hgcons_->waferType(layer, waferU, waferV, true);
+    if (zside < 0)
+      index *= -1;
+    auto itr = occupancy_.find(index);
+    if (itr == occupancy_.end())
+      occupancy_[index] = 1;
+    else
+      ++occupancy_[index];
+    if (type != typef) {
+      auto ktr = badTypes_.find(index);
+      if (ktr == badTypes_.end())
+        badTypes_[index] = 1;
+      else
+        ++badTypes_[index];
+      if (verbosity_ == 1)
+        edm::LogVerbatim("HGCalValidation")
+            << "Detector " << name << " zside = " << zside << " layer = " << layer << " type = " << type << ":" << typef
+            << " wafer = " << waferU << ":" << waferV << " index " << index;
+    }
+    if (verbosity_ > 1)
+      edm::LogVerbatim("HGCalValidation")
+          << "Detector " << name << " zside = " << zside << " layer = " << layer << " type = " << type << ":" << typef
+          << " wafer = " << waferU << ":" << waferV;
+  }
+}
+
+// ------------ method called when starting to processes a run  ------------
+void HGCalWaferHitCheck::beginRun(const edm::Run&, const edm::EventSetup& iSetup) {
+  edm::ESHandle<HGCalDDDConstants> pHGDC;
+  iSetup.get<IdealGeometryRecord>().get(nameDetector_, pHGDC);
+  hgcons_ = &(*pHGDC);
+  if (verbosity_ > 0)
+    edm::LogVerbatim("HGCalValidation") << nameDetector_ << " defined with " << hgcons_->layers(false)
+                                        << " Layers with " << (hgcons_->firstLayer() - 1) << " in front";
+
+  goodChannels_.clear();
+  for (int iz = 0; iz < 2; ++iz) {
+    int zside = iz * 2 - 1;
+    for (int layer = 1; layer <= layerMax; ++layer) {
+      for (int waferU = -waferMax; waferU <= waferMax; ++waferU) {
+        for (int waferV = -waferMax; waferV <= waferMax; ++waferV) {
+          int index = zside * HGCalWaferIndex::waferIndex(layer, waferU, waferV, false);
+          if (hgcons_->isValidHex8(layer, waferU, waferV, true))
+            goodChannels_.emplace_back(index);
+        }
+      }
+    }
+  }
+}
+
+void HGCalWaferHitCheck::endStream() {
+  for (auto itr = occupancy_.begin(); itr != occupancy_.end(); ++itr) {
+    if (globalCache()->occupancy_.find(itr->first) == globalCache()->occupancy_.end())
+      globalCache()->occupancy_[itr->first] = itr->second;
+    else
+      globalCache()->occupancy_[itr->first] += itr->second;
+  }
+  for (auto itr = badTypes_.begin(); itr != badTypes_.end(); ++itr) {
+    if (globalCache()->badTypes_.find(itr->first) == globalCache()->badTypes_.end())
+      globalCache()->badTypes_[itr->first] = itr->second;
+    else
+      globalCache()->badTypes_[itr->first] += itr->second;
+  }
+
+  globalCache()->goodChannels_ = goodChannels_;
```

Comment:
this seems unnecessary; `globalEndJob()` could just use the `goodChannels_` member of the analyzer class
(then the corresponding member in the `Counters` class could just be removed)

Reply:
`goodChannels_` member is not available in `globalEndJob()` because the latter is a static member function.